src/test/lib/optimizer/strategy/between_composition_rule_test.cpp

```diff
@@ -49,13 +49,13 @@ TEST_F(BetweenCompositionTest, ColumnExpressionLeft) {
     PredicateNode::make(less_than_equals_(_a_a, 300),
       _node_a));
 
+  const auto result_lqp = StrategyBaseTest::apply_rule(_rule, input_lqp);
```

Comment:
What made these changes necessary? Not saying I am opposed, just wondering.

Reply:
The new BCR-implementation visits LQPs from the bottom up. If both `input_lqp` and `expected_lqp`, use `_node_a`, the BCR visits and optimizes `expected_lqp` as well. 

By applying the rule before creating `expected_lqp`, we can circumvent this issue.